docs/sources/flow/reference/components/discovery.hetzner.md

```diff
@@ -0,0 +1,171 @@
+---
+canonical: https://grafana.com/docs/agent/latest/flow/reference/components/discovery.hetzner/
+title: discovery.hetzner
+---
+
+# discovery.hetzner
+
+`discovery.hetzner` allows retrieving scrape targets from [Hetzner Cloud API][] and [Robot API][].
```

Comment:
nit: the [Prometheus docs](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#hetzner_sd_config) point out that relabeling exposes the public addresses, and one should use relabeling to get the private one. I'm not familiar with Hetzner, but should we also document something similar?